src/julia-syntax.scm

```diff
@@ -4056,7 +4054,7 @@ f(x) = yt(x)
                                (newlam    (compact-and-renumber (linearize (car exprs)) 'none 0)))
                           `(toplevel-butfirst
                             (block ,@sp-inits
-                                   (method ,name ,(cl-convert sig fname lam namemap defined toplevel interp opaq globals locals)
+                                   (method ,(cadr e) ,(cl-convert sig fname lam namemap defined toplevel interp opaq globals locals)
```

Comment:
I feel skeptical that this is correct, since previously `,name` should always be a symbol that was the syntactic name of the function, and now it is either a symbol or a `!local` (aka !symbol) and we would have to reimplement the method-expr-name pass in C in jl_method_def to figure out what it is

Reply:
Every other branch of this function already does this. The C code completely ignores this field. With the other fix in this PR it doesn't matter, but I wanted to be consistent.

Reply:
Even if this was the 1-arg version of `:method` (for which this argument does something), the code does explicitly contemplate that this could be a GlobalRef: https://github.com/JuliaLang/julia/blob/master/src/interpreter.c#L89-L92

Reply:
ah, okay, I thought we used this except when it was unavailable. Seems to not matter then